homie-controller/tests/integration.rs

```diff
@@ -0,0 +1,220 @@
+    homie
+        .publish_value("node_id", "property_id", 42)
+        .await
+        .unwrap();
+
+    // Wait until the controller receives the value.
+    loop {
+        if let Some(event) = controller.poll(&mut event_loop).await.unwrap() {
+            log::trace!("Event: {:?}", event);
+            if let Event::PropertyValueChanged {
+                device_id,
+                node_id,
+                property_id,
+                value,
+                fresh,
+            } = event
+            {
+                assert_eq!(device_id, "device_id");
+                assert_eq!(node_id, "node_id");
+                assert_eq!(property_id, "property_id");
+                assert_eq!(value, "42");
+                assert_eq!(fresh, true);
+                break;
+            }
+        }
+    }
+
+    // Check that the device looks how we expect.
+    let devices = controller.devices();
+    let device = devices.get("device_id").unwrap();
+    let node = device.nodes.get("node_id").unwrap();
+    let property = node.properties.get("property_id").unwrap();
+    log::info!("Property: {:?}", property);
+    assert_eq!(property.value(), Ok(42));
+
+    // Send a value from the controller to the device.
+    controller
+        .set("device_id", "node_id", "property_id", 13)
+        .await
+        .unwrap();
+
+    // Wait for the device to receive the value and send it back to the controller.
+    loop {
+        if let Some(event) = controller.poll(&mut event_loop).await.unwrap() {
+            log::trace!("Event: {:?}", event);
+            if let Event::PropertyValueChanged {
+                device_id,
+                node_id,
+                property_id,
+                value,
+                fresh,
+            } = event
+            {
+                assert_eq!(device_id, "device_id");
+                assert_eq!(node_id, "node_id");
+                assert_eq!(property_id, "property_id");
+                assert_eq!(value, "13");
+                assert_eq!(fresh, true);
+                break;
+            }
+        }
+    }
+    assert_eq!(*updates.lock().unwrap(), vec!["13".to_string()]);
+
+    // Check that the value sent back is reflected on the controller's view of the device.
+    let devices = controller.devices();
+    let device = devices.get("device_id").unwrap();
+    let node = device.nodes.get("node_id").unwrap();
+    let property = node.properties.get("property_id").unwrap();
+    log::info!("Property: {:?}", property);
+    assert_eq!(property.value(), Ok(13));
+
+    // Disconnect the device.
+    homie.disconnect().await.unwrap();
+    let err = homie_handle.await.unwrap_err();
+    if let SpawnError::Connection(ConnectionError::MqttState(StateError::Io(e))) = err {
+        assert_eq!(e.kind(), ErrorKind::ConnectionAborted);
+    } else {
+        panic!("Unexpected error {:?}", err);
+    }
+}
+
+/// Spawn an MQTT broker listening on the given port.
+fn spawn_mqtt_broker(port: u16) {
```

Comment:
This is very noisy. Can we set up some config/environment to make it shit up?

```
$ cargo test --test integration 
    Finished test [unoptimized + debuginfo] target(s) in 0.25s
     Running target/debug/deps/integration-62f3f2fd7256ed95

running 1 test
 ERROR librumqttd > Disconnected!! Id = 11, Client Id = homie_device
 ERROR librumqttd > Stopped!! Id = 12, Client Id = homie_controller, Network(Io(Os { code: 54, kind: ConnectionReset, message: "Connection reset by peer" }))
test test_device ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s
```

Reply:
Removing the `pretty_env_logger::init()` stops the error logs.